### PR TITLE
OFFENCES-119 Ignore any directly passed child (inchoate) offences for linking/unlinking to schedules. Plus use schedule ref data in tests + some refactoring

### PR DESCRIPTION
**Pull request by yasinmustafa**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/service/OffenceService.kt [hunk 1]
```diff
@@ -27,13 +27,20 @@ class OffenceService(
 ) {
   fun findOffencesByCode(code: String): List<Offence> {
     log.info("Fetching offences by offenceCode")
-    val offences = offenceRepository.findByCodeStartsWithIgnoreCase(code).map { it ->
-      val children = offenceRepository.findByParentOffenceId(it.id)
-      transform(it, children.map { child -> child.id })
-    }
-    return offences.map {
-      it.copy(schedules = transform(offenceSchedulePartRepository.findByOffenceId(it.id)))
+    val offences = offenceRepository.findByCodeStartsWithIgnoreCase(code)
```
Comment (yasinmustafa) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/service/OffenceService.kt:30]: refactored this method to improve performance. Replace calls to the DB in a loop with one call outside the loop

File: src/test/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/resource/ScheduleControllerIntTest.kt [hunk 2]
```diff
@@ -10,115 +10,62 @@ import uk.gov.justice.digital.hmpps.manageoffencesapi.model.SchedulePart
 import uk.gov.justice.digital.hmpps.manageoffencesapi.model.SchedulePartIdAndOffenceId
 
 class ScheduleControllerIntTest : IntegrationTestBase() {
-  @Test
```
Comment (yasinmustafa) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/resource/ScheduleControllerIntTest.kt:13]: Removed this test as the creation of schedules is now done through ref data

File: src/test/resources/test_data/insert-schedule-and-offence-data.sql [hunk 3]
```diff
@@ -14,24 +14,6 @@ VALUES ('XX99002',
         'CJS Title 2Fail to give to an authorised person information / assistance / provide facilities that person may require',
         '2015-03-13', NULL, '2020-06-17 15:31:26.000', '2022-04-07 16:05:58.178', '2022-04-07 16:05:58.178');
 
-INSERT INTO schedule (act, code, url)
```
Comment (yasinmustafa) [inline src/test/resources/test_data/insert-schedule-and-offence-data.sql:17]: schedules are now created through ref data - so this becomes redundant for test data set up

**Review by joelstobart-moj — APPROVED** [review]
LGTM

**Pull request merged**